src/DependencyInjection/Compiler/TemplatingValidatorPass.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+/*
+ * This file is part of the Symfony CMF package.
+ *
+ * (c) 2011-2017 Symfony CMF
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Cmf\Bundle\RoutingBundle\DependencyInjection\Compiler;
+
+use Symfony\Bundle\FrameworkBundle\Templating\EngineInterface;
+use Symfony\Cmf\Bundle\RoutingBundle\Validator\Constraints\RouteDefaultsTemplatingValidator;
+use Symfony\Component\DependencyInjection\Compiler\CompilerPassInterface;
+use Symfony\Component\DependencyInjection\ContainerBuilder;
+
+/**
+ * To avoid a BC-Break: If templating component exists, we will use the validator using general templating
+ * from FrameworkBundle.
+ *
+ * @author Maximilian Berghoff <Maximilian.Berghoff@mayflower.de>
+ */
+class TemplatingValidatorPass implements CompilerPassInterface
+{
+    /**
+     * {@inheritdoc}
+     */
+    public function process(ContainerBuilder $container)
+    {
+        if (class_exists(EngineInterface::class) && $container->has('templating')) {
+            $templatingDefinition = $container->getDefinition('templating');
+            $validatorDefinition = $container->getDefinition('cmf_routing.validator.route_defaults');
+            $validatorDefinition->setClass(RouteDefaultsTemplatingValidator::class);
+            $validatorDefinition->setArgument('$templating', $templatingDefinition);
```

Comment:
does this syntax work with all symfony versions or was that added in 3.2 or so?

Reply:
you can be right, That syntax is new, so i need the index of the argument. Is the counting 0-based?

Reply:
yes, its 0 based. can you check in the sandbox or so if it actually works? i think building a test is a lot of hassle, i am ok if we know it worked at the time we add it...